nautobot/core/models/__init__.py

```diff
@@ -102,6 +102,17 @@ def get_absolute_url(self, api=False):
 
         raise AttributeError(f"Cannot find a URL for {self} ({self._meta.app_label}.{self._meta.model_name})")
 
+    @property
+    def page_title(self):
```

Comment:
Looks like generic test `test_get_object_with_permission` in `core.testing.views` needs to be updated, as right now it's looking for `instance.display` or `str(instance)` as the page title.